scripts/localstorage.js

```diff
@@ -0,0 +1,26 @@
+const form = document.getElementById('contactform');
+let data = {
+  name: '',
+  email: '',
+  message: '',
+};
+
+window.onload = () => {
+  if (localStorage.getItem('data')) {
+    data = JSON.parse(localStorage.getItem('data'));
+    form.name.value = data.name;
+    form.email.value = data.email;
+    form.msg.value = data.message;
+  }
+};
+
+form.addEventListener('submit', () => {
+  data.name = form.name.value;
+  data.email = form.email.value;
+  data.message = form.msg.value;
+  localStorage.setItem('data', JSON.stringify(data));
+});
+
+form.addEventListener('reset', () => {
+  window.localStorage.clear();
+});
```

Comment:
- [ ] Hi @shahedakhatoon, as mentioned by the previous reviewer, your data is not preserved in the local storage. After filling out the form, and not submitting it, the entered values should be saved in the browser. This is very important so that when next you are back on the same page, you can always continue filling out the form from where you ended ie your page should auto-populate with, the values entered in the previous visit to the page. Please let's consider fixing it.

Reply:
@FrederickMih Thanks for the insight, I checked the application storage this time and I think it's working now, I'll submit the changes I've made. Please see the screenshot I've provided.

![Screenshot from 2022-03-12 12-35-04](https://user-images.githubusercontent.com/73285210/158007976-1e0a5d9f-5ab3-438a-bf6e-33ef7bdab524.png)

Reply:
Hi @shahedakhatoon, Nice job so far. :+1: . We can also improve on the form by adding a change event listener on the form input fields so that once the user is entering text, data is immediately preserved in the storage. If you find any difficulties implementing this feature, we can always work together on it. Just hits me on slack. 

I wish you all the best :pray: